libs/vkd3d/swapchain.c

```diff
@@ -385,6 +385,12 @@ static HRESULT d3d12_swapchain_set_display_mode(struct d3d12_swapchain *swapchai
 {
     DXGI_MODE_DESC matching_mode;
     HRESULT hr;
+
+    /* Workaround. FindClosestMatchingMode can take UNKNOWN format, but neither Wine nor DXVK DXGI seems to like it.
+     * Just opt for BGRA8 to workaround the issue. */
+    if (mode->Format == DXGI_FORMAT_UNKNOWN)
+        mode->Format = DXGI_FORMAT_B8G8R8A8_UNORM;
```

Comment:
I think the correct fix here would be to use `swapchain->desc.Format` instead if the given format is `UNKNOWN`. At least that's what DXVK does.

Passing `UNKNOWN` to FindClosestMatchingMode like this is indeed intended to fail.

Reply:
Fixed.